### PR TITLE
Add recipe for docs-versions-menu

### PR DESCRIPTION
**Pull request by goerz**
@conda-forge/help-python This PR creates a *new* feedstock for the [docs-versions-menu](https://pypi.org/project/docs-versions-menu/) project, which was recently renamed from [doctr-versions-menu](https://pypi.org/project/doctr-versions-menu/).

The old project has an active feedstock at https://github.com/conda-forge/doctr-versions-menu-feedstock maintained by @hhslepicka and myself (@goerz); I'm adding both of us as maintainers for the new package as well. Note that I would like to keep the old feedstock active (just like the PyPI entry) for the possibility of making bugfix releases to the old package. 

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by hhslepicka** [comment]
I accept to become a maintainer of this feedstock.
Thanks @goerz

**Comment by hhslepicka** [comment]
@conda-forge/help-python this one is ready to go... could we get a review/merge?

**Comment by goerz** [comment]
@conda-forge/staged-recipes Could someone merge this? Thanks!

File: recipes/docs_versions_menu/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    - packaging
+    - pyparsing >=2.0.2
+    - sphinx
+
+test:
+  imports:
+    - docs_versions_menu
+
+  commands:
+    - docs-versions-menu --help
```
Comment (xhochy) [inline recipes/docs_versions_menu/meta.yaml:38]: Please add a check that all Python requirements are fulfilled.
```suggestion
  requires:
    - pip
  commands:
    - docs-versions-menu --help
    - pip check
```

Comment (goerz) [reply recipes/docs_versions_menu/meta.yaml]: Sure!

**Comment by goerz** [comment]
@xhochy Thanks for the review! I've added the Python requirements check.

**Pull request merged**